### PR TITLE
chore: migrate to core-js

### PR DESCRIPTION
**Pull request by Shyam-Chen**

### CODE REVIEW

File: addon/ng2/blueprints/ng2/files/angular-cli-build.js [hunk 1]
```diff
@@ -12,8 +12,7 @@ module.exports = function(defaults) {
       'systemjs/dist/system-polyfills.js',
```
Comment (kylecordes) [inline addon/ng2/blueprints/ng2/files/angular-cli-build.js]: According to:

https://github.com/systemjs/systemjs

the purpose of system-polyfills is to ensure Promise is available.

Core-js includes a Promise implementation, per:

https://github.com/zloirock/core-js

Therefore, I think this line that copies "system-polyfills.js" should also be removed by this PR.

Comment (Shyam-Chen) [inline addon/ng2/blueprints/ng2/files/angular-cli-build.js]: fix

File: addon/ng2/blueprints/ng2/files/angular-cli-build.js [hunk 2]
```diff
@@ -9,11 +9,9 @@ var Angular2App = require('angular-cli/lib/broccoli/angular2-app');
 module.exports = function(defaults) {
   return new Angular2App(defaults, {
     vendorNpmFiles: [
-      'systemjs/dist/system-polyfills.js',
       'systemjs/dist/system.src.js',
       'zone.js/dist/**/*.+(js|js.map)',
-      'es6-shim/es6-shim.js',
-      'reflect-metadata/**/*.+(ts|js|js.map)',
```
Comment (hansl) [inline addon/ng2/blueprints/ng2/files/angular-cli-build.js]: How is corejs replacing reflect metadata?

**Comment by kylecordes** [comment]
@hansl  According to the core JS page, it include a reflect metadata implementation. You can experimentally verify that it does in fact have them and that Angular 2 works with them.

https://github.com/zloirock/core-js

**Comment by filipesilva** [comment]
Heya @Shyam-Chen, we've just made some pretty big changes in master, can you update?

**Comment by Shyam-Chen** [comment]
@filipesilva I found that the `master` already contains `core-js`.

**Comment by filipesilva** [comment]
Oh nice! I hadn't noticed that but @TheLarkInn added it in as part of the greater webpack move. Thanks for pushing for this feature!

**Pull request closed without merging**